kernel/userspace.c

```diff
@@ -17,6 +17,9 @@
 #include <device.h>
 #include <init.h>
 #include <logging/sys_log.h>
+#ifdef CONFIG_NETWORKING
```

Comment:
You #if-out this include, but I notice that none of your other changes to this file need to live underneath the same #if.  So either this #if isn't necessary or you missed a spot that is going to break when networking is disabled.

Reply:
Or the third option: the header wasn't needed to begin with.

Reply:
This is quite necessary. The auto-generated otype-to-size.h which gets included by this file needs to know the size of a struct net_context if networking is enabled.

I don't include this without an ifdef because you get build errors depending on kernel configuration if you pull this in unconditionally.

Reply:
Then maybe the auto-generated header should include the proper files for its own dependencies?  Or at least document that above the include?  I complain because I'd totally see this kind of thing in the future and rip it out and then waste time trying to figure out what breaks.

Reply:
It's a generated switch body:

static size_t obj_size_get(enum k_objects otype)
{
	switch (otype) {
#include <otype-to-size.h>
	default:
		return sizeof(struct device);
	}
}

It contains:

case K_OBJ_ALERT: return sizeof(struct k_alert);
case K_OBJ_MSGQ: return sizeof(struct k_msgq);
case K_OBJ_MUTEX: return sizeof(struct k_mutex);
case K_OBJ_PIPE: return sizeof(struct k_pipe);
case K_OBJ_QUEUE: return sizeof(struct k_queue);
case K_OBJ_POLL_SIGNAL: return sizeof(struct k_poll_signal);
case K_OBJ_SEM: return sizeof(struct k_sem);
case K_OBJ_STACK: return sizeof(struct k_stack);
case K_OBJ_THREAD: return sizeof(struct k_thread);
case K_OBJ_TIMER: return sizeof(struct k_timer);
#ifdef CONFIG_NETWORKING
case K_OBJ_NET_CONTEXT: return sizeof(struct net_context);
#endif

@lpereira added this to reduce boilerplate when adding new kernel objects.

Reply:
>  Or at least document that above the include? 

Can definitely add some comments explaining what's going on.